runtime/Java/src/org/antlr/v4/runtime/misc/IntegerList.java

```diff
@@ -100,13 +103,9 @@ public final void addAll(IntegerList list) {
 
 	public final void addAll(Collection<Integer> list) {
 		ensureCapacity(_size + list.size());
-		int current = 0;
 		for (int x : list) {
-			_data[_size + current] = x;
-			current++;
-		}
-
-		_size += list.size();
+          _data[_size++] = x;
```

Comment:
:exclamation: Even though it's not documented (and the class is not safe for concurrent operations on multiple threads), the original implementation is saf*er* for single-writer, multiple-reader scenarios. The new implementation contains a sequence point where `list.get(list.size() - 1)` returns an uninitialized value, even if memory fences are employed.